sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/implementation/AmqpReceiveLinkProcessor.java

```diff
@@ -356,23 +369,29 @@ public void request(long request) {
 
         Operators.addCap(REQUESTED, this, request);
 
+        final AmqpReceiveLink link = currentLink;
+        final int credits;
+
         synchronized (creditsAdded) {
-            final AmqpReceiveLink link = currentLink;
-            final int credits = getCreditsToAdd();
+            credits = getCreditsToAdd();
 
-            logger.verbose("linkName[{}] entityPath[{}] request[{}] credits[{}] Backpressure request from downstream.",
-                currentLinkName, entityPath, request, credits);
+            if (link != null && credits > 0) {
+                final boolean hadNoCredits = linkHasNoCredits.compareAndSet(true, false);
```

Comment:
This part is hard to understand. If linkHasNoCredits is successfully set to false, then hasNoCredits is true.

Reply:
I created a variable so it would make it easier to log, but I really don't need the `hadNoCredits` variable.